### PR TITLE
Auto-indent on paste

### PR DESCRIPTION
**Pull request by scholtzan**
This PR adds auto-indentation for pasted text.

## Related Issues
Closes https://github.com/xi-editor/xi-editor/issues/1004

## Review Checklist

- [ ] I have responded to reviews and made changes where appropriate.
- [x] I have tested the code with `cargo test --all` / `./rust/run_all_checks`.
- [x] I have updated comments / documentation related to the changes I made.
- [x] I have rebased my PR branch onto xi-editor/master.

### CODE REVIEW

**Comment by scholtzan** [comment]
Maybe this is also useful for https://github.com/xi-editor/xi-editor/issues/1002

File: rust/syntect-plugin/src/main.rs [hunk 1]
```diff
@@ -50,6 +50,7 @@ type EditBuilder = DeltaBuilder<RopeInfo>;
 pub enum EditType {
     Insert,
     Newline,
+    Other,
```
Comment (cmyr) [inline rust/syntect-plugin/src/main.rs]: really the longer term solution here is to actually expose and use the `EditType` defined in core and use serde

File: rust/syntect-plugin/src/main.rs [hunk 2]
```diff
@@ -73,7 +75,8 @@ struct PluginState {
     // unflushed spans
     spans: Vec<ScopeSpan>,
     new_scopes: Vec<Vec<String>>,
-    indentation_state: Vec<(usize, EditType)>,
+    // keeps track of the lines (start, end) that might need indentation after edit
+    indentation_state: Vec<(usize, usize, EditType)>,
```
Comment (cmyr) [inline rust/syntect-plugin/src/main.rs]: I might start to think about something like,

```rust
enum IndentationTask {
    Newline(usize),
    Edit(usize),
    Batch(Range<usize>),
}
```

**Comment by scholtzan** [comment]
Whoops. I was planning to open another PR for https://github.com/xi-editor/xi-editor/issues/1002 anyway. So I'll do your suggested changes there.

File: rust/syntect-plugin/src/main.rs (hunk 2 above)
Comment (scholtzan) [reply rust/syntect-plugin/src/main.rs]: Looks definitely cleaner

**Comment by cmyr** [comment]
Sure no worries, definitely suggestions more than bugs/problems

**Pull request merged**